docs/ai/conceptual/prompt-engineering-in-dot-net.md

```diff
@@ -0,0 +1,189 @@
+
+Suppose you use an instruction to tell the model to produce a list of presidential accomplishments by category, along with supporting content that tells the model what categories to use. You decide that you want the model to produce a nested list with all caps for categories, with each president's accomplishments in each category listed on one line that begins with their name, with presidents listed chronologically. After your instruction and supporting content, you could add three cues to show the model how to structure and format the list:
+
+```csharp
+prompt = @$"Instructions: Summarize US Presidential accomplishments, grouped by category.
+Categories: Domestic Policy, US Economy, Foreign Affairs, Space Exploration.
+Accomplishments: George Washington
+First president of the United States.
+First president to have been a military veteran.
+First president to be elected to a second term in office.
+First president to receive votes from every presidential elector in an election.
+First president to fill the entire body of the United States federal judges; including the Supreme Court.
+First president to be declared an honorary citizen of a foreign country, and an honorary citizen of France.
+John Adams ...  // Text truncated
+// Cues
+DOMESTIC POLICY
+- George Washington: 
+- John Adams:";
+```
+
+- **DOMESTIC POLICY** shows the model that you want it to start each group with the category in all caps.
+- **- George Washington:** shows the model to start each section with George Washington's accomplishments listed on one line.
+- **- John Adams:** shows the model that it should list remaining presidents in chronological order.
+
+## .NET implementations
+
+This section compares two .NET options for prompt engineering: [Semantic Kernel](/semantic-kernel/overview/) and [Azure OpenAI](/azure/ai-services/openai/overview).
+
+### Semantic Kernel
+
+Semantic Kernel is an SDK that integrates Large Language Models (LLMs) with conventional programming languages such as C#, Python, and Java. Semantic Kernel lets you define plugins and easily chain them together in just a few lines of code. You can use [Semantic Kernel planners](/semantic-kernel/agents/planners/?tabs=Csharp) to automatically orchestrate plugins, which lets you ask an LLM to generate a plan that Semantic Kernel can execute to achieve a user's unique goal.
+
+By deeply integrating with Visual Studio Code, Semantic Kernel also makes it easy for you to integrate [prompt engineering](/semantic-kernel/prompts/your-first-prompt?tabs=Csharp) into your existing development processes:
+
+- Create prompts directly in your preferred code editor.
+- Write tests for your prompts using your existing testing frameworks.
+- Deploy your prompts to production using your existing CI/CD pipelines.
+
+SK Completions use GPT-3 and GPT-3.5 models, which have no specific format rules for prompts. The preceding sections all contain examples of SK Completions.
```

Comment:
Is "SK" a common abbreviation? If so, you should use it in parentheses when you first introduce "Semantic Kernel (SK)". If we prefer customers to use the full words, we should use them in this article.

Reply:
Thanks - I will spell it out for now - I have no idea what we prefer customers to do yet.